controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -118,8 +121,10 @@ func (r *StorageClassRequestReconciler) Reconcile(ctx context.Context, request r
 
 	result, reconcileError = r.reconcilePhases()
 
-	// Apply status changes to the StorageClassRequest
-	statusError := r.Client.Status().Update(r.ctx, r.StorageClassRequest)
+	controllerutil.AddFinalizer(r.StorageClassRequest, v1alpha1.StorageClassRequestFinalizer)
```

Comment:
This should be one of the first steps in the reconcile process - inside reconcilePhases